pkg/distributor/query.go

```diff
@@ -178,130 +178,125 @@ func mergeExemplarQueryResponses(results []interface{}) *ingester_client.Exempla
 	return &ingester_client.ExemplarQueryResponse{Timeseries: result}
 }
 
+type ingesterQueryResult struct {
+	// Why retain the batches rather than build a single slice? We don't need a single slice for each ingester, so building a single slice for each ingester is a waste of time.
+	chunkseriesBatches [][]ingester_client.TimeSeriesChunk
+	timeseriesBatches  [][]mimirpb.TimeSeries
+}
+
 // queryIngesterStream queries the ingesters using the new streaming API.
```

Comment:
[nit] It was "new"... years ago :)

```suggestion
// queryIngesterStream queries the ingesters using the gRPC streaming API.
```